script.js

```diff
@@ -227,3 +227,48 @@ const projectButtons = document.querySelectorAll('.see-post');
 for (let i = 0; i < projectButtons.length; i += 1) {
   projectButtons[i].addEventListener('click', () => { createModal(i); });
 }
+// -----Form-contact---//
+
+const contactForm = document.querySelector('.form-contact');
+
+const email = document.getElementById('user-mail');
+const error = document.querySelector('.error');
+const errormssg = '* The email address should be written in lower case.';
+
+contactForm.addEventListener('submit', (e) => {
+  const flag = email.value.toLowerCase();
+
+  if (flag !== email.value) {
+    e.preventDefault();
+    error.innerText = errormssg;
+  } else {
+    error.innerText = 'succesfully';
+  }
+});
+
+/* -------------preserve data---------*/
+const username = document.getElementById('username');
+
+const message = document.getElementById('message');
+
+function savedata() {
+  const data = {
+    name: username.value,
+    mail: email.value,
+    mssg: message.value,
+  };
+  localStorage.setItem('userInfo', JSON.stringify(data));
+}
+
+username.addEventListener('input', savedata);
+email.addEventListener('input', savedata);
+message.addEventListener('input', savedata);
+
```

Comment:
**[OPTIONAL]**


- Hey, instead of having to repeat yourself for each input, you can add an event listener to form for input, then add the remaining part of your code. Other than that, you've done well. :clap: